compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/model/types/BObjectType.java

```diff
@@ -241,6 +248,10 @@ public boolean isNullable() {
         return false;
     }
 
+    public static void resetAllocatedIds() {
+        DistinctIdSupplier.allocatedIds.clear();
```

Comment:
Here instead of doing this shall we add a map of allocatedIds to the DistinctIdSupplier
```java
private static final Map<Env, Map<BTypeIdSet.BTypeId, Integer>> allocatedIds = Collections.synchronizedMap(new WeakHashMap());

...
private DistinctIdSupplier(Env env) {
      this.env = env;
      allocatedIds.putIfAbscent(env, new ConcurrentHashMap<>());
}
```